### PR TITLE
update fit a line

### PR DESCRIPTION
**Pull request by ceci3**

### CODE REVIEW

File: 01.fit_a_line/README.cn.md [hunk 1]
```diff
@@ -131,30 +137,50 @@ test_reader = paddle.batch(
...
+```python
+filename = 'housing.data'
+feature_name = ['CRIM', 'ZN', 'INDUS', 'CHAS', 'NOX', 'RM', 'AGE',
+               'DIS', 'RAD', 'TAX', 'PTRATIO', 'B', 'LSTAT', 'convert']
+feature_num = len(feature_name)
+data = numpy.fromfile(filename, sep=' ') # 读取原始数据
+data = data.reshape(data.shape[0] // feature_num, feature_num)
+maximums, minimums, avgs = data.max(axis=0), data.min(axis=0), data.sum(axis=0)/data.shape[0]
+
+for i in six.moves.range(feature_num-1):
```
Comment (tink2123) [inline 01.fit_a_line/README.cn.md:155]: six.moves的含义可以在这里添加注释说明吗？

File: 01.fit_a_line/README.cn.md [hunk 2]
```diff
@@ -131,30 +137,50 @@ test_reader = paddle.batch(
...
-x = fluid.layers.data(name='x', shape=[13], dtype='float32')
-y = fluid.layers.data(name='y', shape=[1], dtype='float32')
-y_predict = fluid.layers.fc(input=x, size=1, act=None)
+x = fluid.layers.data(name='x', shape=[13], dtype='float32') # 定义输入的形状和数据类型
+y = fluid.layers.data(name='y', shape=[1], dtype='float32') # 定义输出的形状和数据类型
+y_predict = fluid.layers.fc(input=x, size=1, act=None) # 连接输入和输出的全连接层
 
-main_program = fluid.default_main_program()
-startup_program = fluid.default_startup_program()
+main_program = fluid.[default_main_program](http://www.paddlepaddle.org/documentation/docs/zh/develop/api_cn/fluid_cn.html#default-main-program)()
```
Comment (tink2123) [inline 01.fit_a_line/README.cn.md]: main_program 和 startup_program 这里只能是 python 代码的形式，如果想添加跳转链接的话，试一下把链接添加在上面文字说明位置？或者添加注释说明一下？
当前代码段在jupyter中是如下形式：
default

File: 01.fit_a_line/README.cn.md [hunk 3]
```diff
@@ -131,30 +137,50 @@ test_reader = paddle.batch(
...
 
-在下面的 `SGD optimizer`，`learning_rate` 是训练的速度，与网络的训练收敛速度有关系。
+在下面的 `SGD optimizer`，`learning_rate` 是学习率，与网络的训练收敛速度有关系。
 
 ```python
 sgd_optimizer = fluid.optimizer.SGD(learning_rate=0.001)
 sgd_optimizer.minimize(avg_loss)
 
-#clone a test_program
+#克隆main_program得到test_program，有些operator在训练和测试之间的操作是不同的，例如batch_norm，使用参数for_test来区分该程序是用来训练还是用来测试，该api不会删除任何操作符,请在backward和optimization之前使用。
```
Comment (tink2123) [inline 01.fit_a_line/README.cn.md]: 注释最好不要写在一行内，不方便阅读，尽量分成多行，辛苦啦

File: 01.fit_a_line/README.cn.md [hunk 4]
```diff
@@ -163,26 +189,15 @@ test_program = main_program.clone(for_test=True)
 
 ```python
 use_cuda = False
-place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()
+place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace() # 指明executor的执行场所
 
+###executor可以接受传入的program，并根据feed map(输入映射表)和fetch list(结果获取表)向program中添加数据输入算子和结果获取算子。使用close()关闭该executor，调用run(...)执行program,更多使用详情请参考[fluid.executor](http://www.paddlepaddle.org/documentation/docs/zh/develop/api_cn/fluid_cn.html#permalink-15-executor)
```
Comment (tink2123) [inline 01.fit_a_line/README.cn.md]: 同上，请换行。
且注释里的超链接无法跳转的，请移动到文字部分，或修改一下样式。
当前效果：
default

File: 01.fit_a_line/README.cn.md [hunk 5]
```diff
@@ -234,17 +272,19 @@ for pass_id in range(num_epochs):
             plot_prompt.append(test_prompt, step, test_metics[0])
             plot_prompt.plot()
             # If the accuracy is good enough, we can stop the training.
-            if test_metics[0] < 10.0:
+            if test_metics[0] < 10.0: # 如果准确率达到要求，则停止训练
```
Comment (tink2123) [inline 01.fit_a_line/README.cn.md:294]: 添加中文注释后，可以删除上面的英文了，上同

**Review by tink2123 — COMMENTED** [review]
cool！有一些小的格式问题，可能需要再修改一下~

File: 01.fit_a_line/README.cn.md [hunk 6]
```diff
@@ -264,30 +307,35 @@ inference_scope = fluid.core.Scope()
...
-    for idx, val in enumerate(results[0]):
-        print("%d: %.2f" % (idx, val))
+ print("\nground truth:")
+ for idx, val in enumerate(infer_label):
+     print("%d: %.2f" % (idx, val)) # 打印标签值
 
-    print("\nground truth:")
-    for idx, val in enumerate(infer_label):
-        print("%d: %.2f" % (idx, val))
+save_result(results[0], infer_label) # 保存图片，该函数实现在train.py中
```
Comment (junjun315) [inline 01.fit_a_line/README.cn.md]: md文件会形成jupyter book里的交互页面文件，无法调用train.py里的函数。如果需要使用，请在当前的页面里添加该函数。

File: 01.fit_a_line/README.cn.md [hunk 7]
```diff
@@ -163,88 +194,100 @@ test_program = main_program.clone(for_test=True)
...
-    if params_dirname is not None:
-        # We can save the trained parameters for the inferences later
-        fluid.io.save_inference_model(params_dirname, ['x'],
-                                      [y_predict], exe)
+```
+
+保存训练参数到之前给定的路径中
+```python
+if params_dirname is not None:
+    fluid.io.save_inference_model(params_dirname, ['x'], [y_predict], exe)
```
Comment (junjun315) [inline 01.fit_a_line/README.cn.md]: 保存模型的位置不对。 这里已经出了训练循环，有些参数取得可能不对

File: 01.fit_a_line/README.cn.md [hunk 8]
```diff
@@ -163,88 +194,100 @@ test_program = main_program.clone(for_test=True)
...
 naive_exe.run(startup_program)
 step = 0
 
 exe_test = fluid.Executor(place)
+```
 
-# main train loop.
+paddlepaddle提供了reader机制来读取训练数据。reader会一次提供多列数据，因此我们需要一个python的列表来定义读取顺序。我们构建一个循环来进行训练，直到训练结果足够好或者循环次数足够多。
+如果训练迭代次数满足参数保存的迭代次数，可以把训练参数保存到`params_dirname`。
+其次设置训练主循环
```
Comment (junjun315) [inline 01.fit_a_line/README.cn.md]: “其次”设置训练主循环这里，其次有点突兀。虽然知道是跟前面的“首先”对应，但是距离较远，也没有分割。可以试试看换行或者设置不同的级次来关联上下文

File: 01.fit_a_line/README.cn.md [hunk 9]
```diff
@@ -131,30 +137,55 @@ test_reader = paddle.batch(
         batch_size=BATCH_SIZE)
 ```
 
+如果想直接从txt文件中读取数据的话，可以参考以下代码。
+
+```python
+filename = 'housing.data'
```
Comment (junjun315) [inline 01.fit_a_line/README.cn.md]: 如果我们提供了示例，是否需要把housing.data文件放上来或者提供下载路径？否则可能用户也不知道文件的格式

File: 01.fit_a_line/README.cn.md [hunk 10]
```diff
@@ -131,30 +137,55 @@ test_reader = paddle.batch(
...
+data = data.reshape(data.shape[0] // feature_num, feature_num)
+maximums, minimums, avgs = data.max(axis=0), data.min(axis=0), data.sum(axis=0)/data.shape[0]
+
+for i in six.moves.range(feature_num-1): # six.moves可以兼容python2和python3
+    data[:, i] = (data[:, i] - avgs[i]) / (maximums[i] - minimums[i])
+
+ratio = 0.8 # 训练集和验证集的划分比例
+offset = int(data.shape[0]*ratio)
+train_data = data[:offset]
+test_data = data[offset:]
```
Comment (junjun315) [inline 01.fit_a_line/README.cn.md:164]: 这样处理完了train_data和test_data后，下面的训练过程并不能直接使用吧？

**Review by junjun315 — COMMENTED** [review]
还有一些问题，请帮忙看看是否需要修改

File: 01.fit_a_line/README.cn.md (hunk 6 above)
Comment (ceci3) [reply 01.fit_a_line/README.cn.md]: 已修改

File: 01.fit_a_line/README.cn.md (hunk 9 above)
Comment (ceci3) [reply 01.fit_a_line/README.cn.md]: 数据集下载路径之前就有了

File: 01.fit_a_line/README.cn.md (hunk 10 above)
Comment (ceci3) [reply 01.fit_a_line/README.cn.md:164]: 已经完善

**Review by junjun315 — APPROVED** [review]
LGTM
